File: server/makc2022--nodejs--nestjs/src/layer1/common/common-environment.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { Injectable } from '@nestjs/common';

/** Общее окружение. */
@Injectable()
export class CommonEnvironment {
  /** Базовый путь: абсолютный путь к папке, относительно которой указываются пути к файлам. */
  basePath = __dirname;

  /** Имя. */
  name = process.env.NODE_ENV ?? 'development';

  /** Получить признак окружения для разработки.
   * @returns Признак окружения для разработки.
   */
  isDevelopment(): boolean {
    return this.name === 'development';
  }

  /** Получить признак продуктового окружения.
   * @returns Признак продуктового окружения.
   */
  isProduction(): boolean {
    return this.name === 'production';
  }

  /** Получить признак окружения для тестирования.
   * @returns Признак окружения для тестирования.
   */
  isTest(): boolean {
    return this.name === 'test';
  }
}

File: server/makc2022--dotnet/Makc2022.Layer1/Common/CommonEnvironment.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Makc2022.Layer1.Common
{
    /// <summary>
    /// Общее окружение.
    /// </summary>
    public class CommonEnvironment
    {
        #region Properties

        /// <summary>
        /// Базовый путь: абсолютный путь к папке, относительно которой указываются пути к файлам.
        /// </summary>
        public string BasePath { get; set; } = AppContext.BaseDirectory;

        /// <summary>
        /// Имя.
        /// </summary>
        public string? Name { get; set; } = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");

        #endregion Properties
    }
}
